File: kattis/training1/woodcutting.java
import java.util.*;

public class woodcutting {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        for (int i = 0; i < t; i++) {
            int c = sc.nextInt();
            long[] custs = new long[c];

            for (int j = 0; j < c; j++) {
                long tot = sc.nextLong();
                for (int k = 0; k < tot; k++) {
                    custs[j] += sc.nextLong();
                }
            }

            Arrays.sort(custs);

            double time = 0.0;
            double total = 0.0;

            for (int j = 0; j < c; j++) {
                time += custs[j];
                total += time;
            }

            System.out.println(total / c);
        }
    }
}

File: kattis/training1/woodcutting.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#define ll long long 

using namespace std;

int main() {
    ll t; cin >> t;
    for (ll i = 0; i < t; i++) {
        ll c; cin >> c;
        vector<ll> custs;
        for (ll j = 0; j < c; j++) {
            ll num; cin >> num;
            ll sum = 0;
            for (ll k = 0; k < num; k++) {
                ll y; cin >> y;
                sum += y;
            }

            custs.push_back(sum);
        }

        sort(custs.begin(), custs.end());

        ll time = 0;
        ll total = 0;
        for (ll j = 0; j < c; j++) {
            time += custs[j];
            total += time;
        }

        cout << total * 1.0 / c << '\n';
    }

    return 0;
}

